addons/payment/tests/test_payment_provider.py

```diff
@@ -226,3 +226,32 @@ def test_provider_not_compatible_with_express_checkout(self):
             self.company.id, self.partner.id, self.amount, is_express_checkout=True
         )
         self.assertNotIn(self.provider, compatible_providers)
+
+    def test_payment_provider_availability_report(self):
+        """ Test that report values are correctly set when the provider is available and
+        unavailable. """
+
+        report = {}
+        self.env['payment.provider']._get_compatible_providers(
+            self.company.id, self.partner.id, self.amount, report=report
+        )
+
+        # Test that report is configured correctly for available payment provider.
+        self.assertIn('providers', report)
+        self.assertIn(self.provider, report['providers'])
+        self.assertTrue(report['providers'][self.provider]['available'])
+        self.assertFalse(report['providers'][self.provider]['reason'])
+
+        # Test that for incompatible provider 'available' is set to False and the reason is defined.
```

Comment:
```suggestion
        # Test that for an incompatible provider, 'available' is set to False and the reason is defined.
```

Reply:
Comment resolved without change

Reply:
Oops, :see_no_evil: , thanks for the remind!